lib/streamlit/report_session.py

```diff
@@ -417,6 +399,34 @@ def _enqueue_report_finished_message(self, status):
         msg.report_finished = status
         self.enqueue(msg)
 
+    def handle_git_information_request(self):
+        msg = ForwardMsg()
+
+        try:
+            from streamlit.git_util import GitRepo
+
+            self._repo = GitRepo(self._report.script_path)
+
+            repo, branch, module = self._repo.get_repo_info()
+
+            msg.git_info_changed.repository = repo
+            msg.git_info_changed.branch = branch
+            msg.git_info_changed.module = module
+
+            msg.git_info_changed.untracked_files[:] = self._repo.untracked_files
+            msg.git_info_changed.uncommitted_files[:] = self._repo.uncommitted_files
+
+            if self._repo.is_head_detached:
+                msg.git_info_changed.state = GitInfo.GitStates.HEAD_DETACHED
+            elif len(self._repo.ahead_commits) > 0:
+                msg.git_info_changed.state = GitInfo.GitStates.AHEAD_OF_REMOTE
+            else:
+                msg.git_info_changed.state = GitInfo.GitStates.DEFAULT
+        except:
```

Comment:
Hm, are we okay with swallowing all errors here? Seems a bit dangerous to do so as we may end up letting silent failures get through and only surface as bugs down the line where they're harder to diagnose.

We could probably achieve similar behavior without being worried about letting problematic silent failures slip through by checking if `self._repo.is_valid()` first, then skipping over the stuff that will definitely explode if it's `False`. Even if we don't want to truly raise an error here when things break, I think we'd probably at least want to log an error or warning.

It also seems weird to me that in the case where an error does happen we send an empty/mostly empty message to the client, but that might be fine if the client handles that case properly.

Reply:
The truth is we don't expect users to have Git or a specific version, and that causes errors we'd need to classify, we basically want to let anything slide since it's an optional feature for users. I addressed this by putting a debug warning. This helps if the Git Deploy somehow should be working, but isn't we can get the error from there.